Lambdas/GHHooks/Extensions/Embed+Equtable.swift

```diff
@@ -42,13 +46,30 @@ extension Embed.Author: Equatable {
     public static func == (lhs: Embed.Author, rhs: Embed.Author) -> Bool {
         lhs.url == rhs.url &&
         lhs.icon_url?.asString == rhs.icon_url?.asString &&
-        lhs.name == rhs.name
+        lhs.name == rhs.name &&
+        anyNilOrEqual(lhs.proxy_icon_url, rhs.proxy_icon_url)
     }
 }
 
 extension Embed.Field: Equatable {
     public static func == (lhs: Embed.Field, rhs: Embed.Field) -> Bool {
         lhs.name == rhs.name &&
-        lhs.value == rhs.value
+        lhs.value == rhs.value &&
+        anyNilOrEqual(lhs.inline, rhs.inline)
+    }
+}
+
+/// Returns `true` if both values are nil, or if both values are the same. Otherwise `false`.
+/// This is used for the fields that even if we send `nil` for, Discord might populate them itself.
+private func anyNilOrEqual<E: Equatable>(_ lhs: E?, _ rhs: E?) -> Bool {
+    switch (lhs, rhs) {
+    case (.none, _):
+        return true
+    case (_, .none):
+        return true
+    case let (.some(lhs), .some(rhs)):
+        return lhs == rhs
+    default:
+        fatalError("Impossible")
     }
```

Comment:
1. The comment is wrong - it effectively says "true if both are nil or both the same, false if only one is nil or they're unequal", but the actual semantics are "true unless both are non-nil _and_ unequal".
2. This is a _very_ rare case where I think a custom operator would actually be appropriate:
    ```swift
    infix operator ?==: ComparisonPrecedence
    
    @_transparent
    public func ?== <T: Equatable>(lhs: T?, rhs: T?) -> Bool {
        (lhs ?? rhs) == (rhs ?? lhs)
    }
    ```
    (How's that for a confusing-looking way to provide the desired semantics? 😁)

Reply:
Hmm i'd rather keep the switch just so it doesn't result in unnecessary equitability checks.